### PR TITLE
Puppet Dev Docs

### PR DESCRIPTION
**Pull request by rambleraptor**
If everybody likes the format of this, I'll build a version for Chef + Ansible.

-----------------------------------------------------------------
# [all]
Puppet Dev Docs

### CODE REVIEW

File: docs/providers/puppet.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+### Placing Puppet Modules into proper location.
+Puppet modules can be installed from Supermarket using the following command:
+```
+   puppet module install google-gcompute
+```
+In many cases, you'll want to be running these modules from source.
+
+All Puppet modules are located in `~/.puppetlabs/etc/code/modules`. You'll have
+to place each module inside this folder using the naming convention
+`g<product name`. No underscores are necessary.
```
Comment (ndmckinley) [inline docs/providers/puppet.md]: nit, you probably meant `<product name>`

File: docs/providers/puppet.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  > google-api-client (0.15.0)
+  > googleauth (0.5.3)
+```
+
+## Running a Puppet Example
+```
+  FACTER_cred_path=<path to service account> puppet apply <path to example>
+```
+All environment variables have to be passed in with the `FACTER_` prefix.
+You may have to change project names on examples.
```
Comment (ndmckinley) [inline docs/providers/puppet.md]: Maybe add in `sed -i 's/google.com:graphite-playground/your-project/g' path/to/example`.

File: docs/providers/puppet.md (hunk 1 above)
Comment (rambleraptor) [reply docs/providers/puppet.md]: Done!

File: docs/providers/puppet.md (hunk 2 above)
Comment (rambleraptor) [reply docs/providers/puppet.md]: Great note. Added

**Pull request merged**